includes/class-wcsatt-scheme-prices.php

```diff
@@ -1,388 +1,484 @@
+				$prices[ 'sale_price' ]    = $subscription_scheme->get_sale_price();
 
-				$prices_array = array(
-					'price'         => $prices,
-					'regular_price' => $regular_prices,
-					'sale_price'    => $sale_prices
-				);
+				$prices[ 'price' ] = '' !== $prices[ 'sale_price' ] && $prices[ 'sale_price' ] < $prices[ 'regular_price' ] ? $prices[ 'sale_price' ] : $prices[ 'regular_price' ];
+
+			} elseif ( 'inherit' === $subscription_scheme->get_pricing_mode() && $subscription_scheme->get_discount() > 0 && $raw_prices[ 'price' ] > 0 ) {
+
+				$prices[ 'regular_price' ] = empty( $prices[ 'regular_price' ] ) ? $prices[ 'price' ] : $prices[ 'regular_price' ];
+				$prices[ 'price' ]         = self::get_discounted_scheme_price( $raw_prices, $subscription_scheme->get_discount() );
+
+				if ( $prices[ 'price' ] < $prices[ 'regular_price' ] ) {
+					$prices[ 'sale_price' ] = $prices[ 'price' ] ;
+				}
 			}
 		}
 
-		return $prices_array;
+		return apply_filters( 'wcsatt_subscription_scheme_prices', $prices, $subscription_scheme );
 	}
 
 	/**
-	 * Filter get_price() calls to take price overrides into account.
+	 * Indicates whether the product price is modified by one or more subscription schemes.
 	 *
-	 * @param  double       $price      unmodified price
-	 * @param  WC_Product   $product    the bundled product
-	 * @return double                   modified price
+	 * @param  array  $subscription_schemes
+	 * @return boolean
 	 */
-	public static function filter_get_price( $price, $product ) {
+	public static function price_filter_exists( $subscription_schemes ) {
 
-		$subscription_scheme = self::$price_overriding_scheme;
+		$price_filter_exists = false;
 
-		if ( $subscription_scheme ) {
-
-			if ( apply_filters( 'wcsatt_price_filters_allowed', true, self::$price_overridden_product, $subscription_scheme, $product ) ) {
-
-				$prices_array = array(
-					'price'         => $price,
-					'regular_price' => $product->regular_price,
-					'sale_price'    => $product->sale_price
-				);
-
-				$overridden_prices = self::get_subscription_scheme_prices( $prices_array, $subscription_scheme );
-				$price             = $overridden_prices[ 'price' ];
+		foreach ( $subscription_schemes as $subscription_scheme ) {
+			if ( $subscription_scheme->has_price_filter() ) {
+				$price_filter_exists = true;
+				break;
 			}
 		}
 
-		return $price;
+		return $price_filter_exists;
 	}
 
 	/**
-	 * Filter get_regular_price() calls to take price overrides into account.
+	 * Returns the "base" subscription scheme by finding the one with the lowest recurring price.
+	 * If prices are equal, no interval-based comparison is carried out:
+	 * Reason: In some applications "$5 every week for 2 weeks" (=$10) might be seen as "cheaper" than "$5 every month for 3 months" (=$15), and in some the opposite.
+	 * Instead of making guesswork and complex calculations, we can let scheme order be used to define the "base" scheme manually.
```

Comment:
Since the "scheme options" admin UI allows users to sort them, I thought it might be best to let sort order imply which scheme is the "base" one when all recurring prices are equal.

Note sure how/if we should revise this when we introduce trials and sign up fees. Personally I'd keep it simple and still rely on sort order to define the "base".